### PR TITLE
Rename coin solution to coin spend

### PR DESCRIPTION
**Pull request by richardkiss**

### CODE REVIEW

**Comment by aqk** [comment]
Yay! Thanks for making this more understandable!

File: chia/types/coin_solution.py [hunk 1]
```diff
@@ -1,24 +1,6 @@
...
-    more often that it is streamed.
-    """
-
-    coin: Coin
-    puzzle_reveal: SerializedProgram
-    solution: SerializedProgram
-
-    def additions(self) -> List[Coin]:
-        return additions_for_solution(self.coin.name(), self.puzzle_reveal, self.solution, INFINITE_COST)
+warnings.warn("`CoinSolution` is now `CoinSpend`")
```
Comment (aqk) [inline chia/types/coin_solution.py:6]: Cool - this is good to know about.

**Review by aqk — DISMISSED** [review]
Thanks for doing that. I checked that no .clvm.hex files changed, also.

Looks good.

**Review by wjblanke — APPROVED** [review]
aok

**Comment by AlexSSD7** [comment]
@richardkiss @aqk @wjblanke 

THIS PR BROKE THE API. Spend bundle in `push_tx` key was `coin_solutions` before this, and was changed to `coin_spends` WITHOUT ANY PRIOR NOTICE or any backward compatibility whatsoever.

**Pull request merged**